processing/src/main/java/org/apache/druid/segment/data/FrontCodedIndexedWriter.java

```diff
@@ -413,15 +413,62 @@ private static int compareNullableUtf8UsingJavaStringOrdering(
     return StringUtils.compareUtf8UsingJavaStringOrdering(b1, b2);
   }
 
+  /**
+   * Copy of instance method {@link FrontCodedIndexed.FrontCodedV0#getFromBucket(ByteBuffer, int)}
```

Comment:
Why does this need to be a copy?

Reply:
i guess I could keep it static and have the instance method on the reader just directly call the shared static method since v0 reader doesn't have any need to be an instance method.

The V1 writer is like this with its own copy because the reader side of it has to be an instance method because it keeps some re-used int arrays to store buffer positions and fragment lengths help with rebuilding the values, so I think i was just thinking i would do the same thing and have a writer copy for v0 to skip any potential overhead of the static method call and be more consistent with v1.

Reply:
reverted to use static method